projects/core/src/lib/layers/layer-list/layer-list-element/layer-list-element.component.ts

```diff
@@ -185,4 +186,8 @@ export class LayerListElementComponent {
     showRasterDownload(layer: Layer): void {
         this.layoutService.setSidenavContentComponent({component: DownloadRasterLayerComponent, config: {layer}});
     }
+
+    showVectorDownload(layer: Layer): void {
+        this.layoutService.setSidenavContentComponent({component: DownloadVectorLayerComponent, config: {layer}});
+    }
```

Comment:
Couldn't you have just one method showDownload?
It takes the layer, then distinguishes its type, and opens the correct dialog.

This would have also the advantage that the HTML file would only need one button, regardless of the layer type.

Reply:
https://github.com/geo-engine/geoengine-ui/pull/722/commits/937713ca40b7a44c1e1bae00a77645322c63bc33